keras/utils/vis_utils.py

```diff
@@ -149,7 +149,7 @@ def model_to_dot(model,
                 inputlabels = str(layer.input_shape)
             elif hasattr(layer, 'input_shapes'):
                 inputlabels = ', '.join(
-                    [str(ishape) for ishape in layer.input_shapes])
```

Comment:
Note that it's not the same to use `()` or `[]`. One is a generator and the other one is a list. A list uses more memory and is slower. So the change you did isn't needed.

Reply:
This was the exactly change I did, actually, changing from a list to a generator to decrease the memory usage and increase performance.

Reply:
Woops my bad